.github/actions/setup-macos-builder/action.yaml

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Prepare Builder for MacOS
+description: 'Prepare Build Environment'
+inputs:
+  rust-version:
+    description: 'version of rust to install (e.g. nightly)'
+    required: true
+    default: 'nightly'
+  jdk-version:
+    description: 'jdk version to install (e.g., 17)'
+    required: true
+    default: '17'
+  architecture:
+    description: 'OS architecture for the JDK'
+    required: true
+    default: 'x64'
+runs:
+  using: "composite"
+  steps:
+    - name: Install Build Dependencies
+      shell: bash
+      run: |
+        # Install protobuf
+        mkdir -p $HOME/d/protoc
+        cd $HOME/d/protoc
+        export PROTO_ZIP="protoc-21.4-osx-aarch_64.zip"
```

Comment:
Maybe we can use `architecture` here?

Reply:
hmm yea, I don't know how it worked for Mac `x86_64` 😂 .

Reply:
Same as DataFusion experience. 😂